llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -64,6 +65,50 @@ cl::opt<RelEncodeLoc> RelocLocation(
                clEnumVal(CompileUnit, "In compile unit")),
     cl::init(Atom));
 
+static bool isScatteredRelocation(const MachO::any_relocation_info &RelocInfo,
+                                  Triple::ArchType Arch) {
+  if (Arch == Triple::ArchType::x86_64)
+    return false;
+  return RelocInfo.r_word0 & MachO::R_SCATTERED;
+}
+
+static uint64_t getRelocationType(const MachO::any_relocation_info &RelocInfo,
+                                  support::endianness Endian) {
+  if (Endian == support::little ||
+      (Endian == support::native &&
```

Comment:
There is no reason to check against native endianness since that is wrong since we can be cross compiling. You can assert on native endianness because you only care about object endianness.